float64[] current_coordinates
float64[] desired_coordinates